nnfabrik/main.py

```diff
@@ -69,24 +125,31 @@ def add_entry(
         self,
         model_fn,
         model_config,
-        model_fabrikant=None,
         model_comment="",
+        model_fabrikant=None,
         skip_duplicates=False,
+        return_key_only=True,
     ):
         """
         Add a new entry to the model.
 
         Args:
-            model_fn (string) - name of a callable object. If name contains multiple parts separated by `.`, this is assumed to be found in a another module and
+            model_fn (str, Callable): name of a callable object. If name contains multiple parts separated by `.`, this is assumed to be found in a another module and
                 dynamic name resolution will be attempted. Other wise, the name will be checked inside `models` subpackage.
-            model_config (dict) - Python dictionary containing keyword arguments for the model_fn
-            model_fabrikant (string) - The fabrikant name. Must match an existing entry in Fabrikant table. If ignored, will attempt to resolve Fabrikant based on the database user name for the existing connection.
+            model_config (dict): Python dictionary containing keyword arguments for the model_fn
             model_comment - Optional comment for the entry.
-            skip_duplicates - If True, no error is thrown when a duplicate entry (i.e. entry with same model_fn and model_config) is found.
+            model_fabrikant (str): The fabrikant name. Must match an existing entry in Fabrikant table. If ignored, will attempt to resolve Fabrikant based on the database user name for the existing connection.
+            skip_duplicates (bool, optional): If True, no error is thrown when a duplicate entry (i.e. entry with same model_fn and model_config) is found. Defaults to False.
+            return_key_only (bool, optional): If True, only the primary key attribute for the new entry or corresponding existing entry is returned. Otherwise, the entire 
+                entry is returned. Defaults to True.
 
         Returns:
-            key - key in the table corresponding to the entry.
+            dict: the entry in the table corresponding to the new (or possibly existing, if skip_duplicates=True) entry.
         """
+        if not isinstance(model_fn, str):
+            # infer the full path to the callable
+            model_fn = model_fn.__module__ + "." + model_fn.__name__
```

Comment:
Just a question: why do we need to get the path if we have a callable?

Reply:
because we can only register the path into the table, and we have to verify that the path that can be guessed is a valid one. For example, you could just create a callable in the interpreter and that unfortunately will not have a resolvable path with it.